src/Analyzers/CSharp/Tests/UseConditionalExpression/UseConditionalExpressionForAssignmentTests.cs

```diff
@@ -2186,4 +2186,244 @@ void M(bool containsHighBits)
             }
             """, LanguageVersion.CSharp9);
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/63441")]
+    public async Task TestNullCheck1()
+    {
+        await TestMissingAsync("""
+            using System;
+            public class Program
+            {
+                public static void TestMethod(Test test)
+                {
+                    if (test != null && test.Field == null)
+                    {
+                        test.Field = string.Empty;
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException();
+                    }
+                }
+                public class Test
+                {
+                    public string Field;
+                }
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/63441")]
+    public async Task TestNullCheck1_B()
+    {
+        await TestMissingAsync("""
+            using System;
+            public class Program
+            {
+                public static void TestMethod(Test test)
+                {
+                    if (null != test && test.Field == null)
+                    {
+                        test.Field = string.Empty;
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException();
+                    }
+                }
+                public class Test
+                {
+                    public string Field;
+                }
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/63441")]
+    public async Task TestNullCheck2()
+    {
+        await TestMissingAsync("""
+            using System;
+            public class Program
+            {
+                public static void TestMethod(Test test)
+                {
+                    if (test is not null && test.Field is null)
```

Comment:
>if (test is not null && test.Field is null) [](http://example.com/codeflow?start=20&length=43)

Is something like the following handled?

if (test?.Field is not null)
{
    test.Field = string.Empty;
}
else
{
    throw new InvalidOperationException();
}

Reply:
probably not.  i can see about handling that in followup.